spring-integration-core/src/main/java/org/springframework/integration/handler/AbstractMessageHandler.java

```diff
@@ -131,36 +141,44 @@ public void handleMessage(Message<?> message) {
 		MetricsContext start = null;
 		boolean countsEnabled = this.countsEnabled;
 		AbstractMessageHandlerMetrics handlerMetrics = this.handlerMetrics;
+		Sample sample = null;
+		if (countsEnabled && this.meterRegistry != null) {
+			sample = Timer.start(this.meterRegistry);
+		}
 		try {
 			if (this.shouldTrack) {
 				message = MessageHistory.write(message, this, this.getMessageBuilderFactory());
 			}
 			if (countsEnabled) {
-				if (handlerMetrics.getTimer() != null) {
-					final Message<?> messageToSend = message;
-					handlerMetrics.getTimer().recordCallable(() -> {
-						handleMessageInternal(messageToSend);
-						return null;
-					});
-				}
-				else {
-					start = handlerMetrics.beforeHandle();
-					handleMessageInternal(message);
-					handlerMetrics.afterHandle(start, true);
+				start = handlerMetrics.beforeHandle();
+				handleMessageInternal(message);
+				if (this.meterRegistry != null) {
+					sample.stop(Timer.builder(SEND_TIMER_NAME)
+							.tag("type", "handler")
+							.tag("name", getComponentName() == null ? "unknown" : getComponentName())
+							.tag("result", "success")
+							.tag("exception", "none")
+							.description("Subflow process time")
+							.register(this.meterRegistry));
 				}
+				handlerMetrics.afterHandle(start, true);
```

Comment:
The thought is that Micrometer metrics and legacy metrics shouldn't be mutually exclusive. Since Boot 2.0 is about to ship with an autoconfigured `MeterRegistry` in `spring-boot-actuator-autoconfigure`, making them mutually exclusive like it was in s-i 5.0.2 would force the end user to make a choice between `spring-boot-actuator` and legacy metrics in `spring-integration`.

Reply:
So, that's a big unfortunate that we have released `5.0 GA` far early before Boot `2.0 GA`...

Too much breaking changes in the point release... 

:cry:

Reply:
What do you see as a breaking change?